financialaid/views.py

```diff
@@ -51,6 +56,35 @@ def get_queryset(self):  # pragma: no cover
         return None
 
 
+class FinancialAidSkipView(UpdateAPIView):
+    """
+    View for financial aid skip API. Takes user and program, then determines whether a financial
+    aid object exists, and then either creates or updates a financial aid object to reflect
+    the user skipping financial aid.
+    """
+    serializer_class = FinancialAidSkipSerializer
+    authentication_classes = (SessionAuthentication, )
+    permission_classes = (IsAuthenticated, )
+
+    def get_object(self):  # pragma: no cover
+        """
+        Overrides get_object in case financialaid object does not exist, as the learner may skip
+        financial aid either after starting the process or in lieu of applying
+        """
+        program = get_object_or_404(Program, id=self.kwargs["program_id"])
+        if not program.financial_aid_availability:
+            raise ValidationError("Financial aid not available for this program.")
+        if not ProgramEnrollment.objects.filter(program=program.id, user=self.request.user).exists():
+            raise ValidationError("User not in program.")
+        tier_program = get_no_discount_tier_program(program.id)
```

Comment:
`get_no_discount_tier_program` can raise `DoesNotExist`.  We should wrap that call inside `get_no_discount_tier_program` to return a `ImproperlyConfigured` exception in this case.